src/api/app/policies/token/release_policy.rb

```diff
@@ -1,7 +1,17 @@
 class Token::ReleasePolicy < TokenPolicy
   def trigger?
     return false unless user.is_active?
+    return false unless sufficient_permission_on_all_release_targets?
 
     PackagePolicy.new(user, record.object_to_authorize).update?
   end
+
+  private
+
+  # we need to check all release targets upfront to avoid incomplete releases
+  def sufficient_permission_on_all_release_targets?
+    record.object_to_authorize.project.release_targets.where(trigger: 'manual').each do |release_target|
+      return false unless user.can_modify?(release_target.target_repository.project)
```

Comment:
For this authorization check, we should also use Pundit with the `ProjectPolicy#update?`: https://github.com/openSUSE/open-build-service/blob/a37868ae778e212287d1e5ae6514857699396360/src/api/app/policies/project_policy.rb#L12

Reply:
@dmarcoux good point, done